### PR TITLE
Fix encoding bug with python3 setup.py

### PR DESCRIPTION
**Pull request by jmoldow**
In setup.py running with python3, ``open('README.rst').read()`` can fail with UnicodeDecodeError: 'ascii' codec can't decode byte.

In python3, `open()` takes an optional encoding argument; when not specified, ``locale.getpreferredencoding(False)`` is used to get the current locale encoding. This may be 'ascii', which is the incorrect encoding for rotunicode's README, which contains non-ASCII characters.

Fix this problem by using `codecs.open()` instead. This function takes an encoding argument on all supported versions of python, whereas the builtin `open()` function only accepts an encoding
argument in python3.

Test for this problem with tox by having requirements.txt (and therefore, also requirements-dev.txt) specify this project as a requirement for install.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jmoldow has signed the CLA. Thanks for the pull request!

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 six
+-e .
```
Comment (Jeff-Meadows) [inline requirements.txt]: ?

**Comment by Jeff-Meadows** [comment]
lgtm besides the addition to requirements.txt - not sure what that's for.

File: requirements.txt (hunk 1 above)
Comment (jmoldow) [inline requirements.txt]: > Test for this problem with tox by having requirements.txt (and therefore, also requirements-dev.txt) specify this project as a requirement for install.

**Pull request merged**